go/test/endtoend/vtgate/schema/schema_test.go

```diff
@@ -0,0 +1,290 @@
+	err := ioutil.WriteFile(sqlFile, []byte("create table test_table_x (id int)"), 0644)
+	assert.Nil(t, err)
+	timeout := time.Now().Add(10 * time.Second)
+	matchFoundAfterAutoSchemaApply := false
+	for time.Now().Before(timeout) {
+		if _, err := os.Stat(sqlFile); os.IsNotExist(err) {
+			matchFoundAfterAutoSchemaApply = true
+			checkTables(t, totalTableCount+1)
+			matchSchema(t, clusterInstance.Keyspaces[0].Shards[0].Vttablets[0].VttabletProcess.TabletPath, clusterInstance.Keyspaces[0].Shards[1].Vttablets[0].VttabletProcess.TabletPath)
+		}
+	}
+	if !matchFoundAfterAutoSchemaApply {
+		assert.Fail(t, "Auto schema is not consumed")
+	}
+	defer os.RemoveAll(path.Join(schemaChangeDirectory, keyspaceName))
+}
+
+// matchSchema schema for supplied tablets should match
+func matchSchema(t *testing.T, firstTablet string, secondTablet string) {
+	firstShardSchema, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("GetSchema", firstTablet)
+	assert.Nil(t, err)
+
+	secondShardSchema, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("GetSchema", secondTablet)
+	assert.Nil(t, err)
+
+	assert.Equal(t, firstShardSchema, secondShardSchema)
+}
+
+// testSchemaChangePreflightErrorPartialy applying same schema + new schema should throw error for existing one
+// Tests that some SQL statements fail properly during PreflightSchema.
+func testSchemaChangePreflightErrorPartialy(t *testing.T) {
+	createNewTable := fmt.Sprintf(createTable, fmt.Sprintf("vt_select_test_%02d", 5)) + fmt.Sprintf(createTable, fmt.Sprintf("vt_select_test_%02d", 2))
+	output, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("ApplySchema", "-sql", createNewTable, keyspaceName)
+	assert.NotNil(t, err)
+	assert.True(t, strings.Contains(output, "already exists"))
+
+	checkTables(t, totalTableCount)
+}
+
+// testSchemaChangePreflightErrorPartialy applying same schema + new schema should throw error for existing one and also add the new schema
+//Tests the PreflightSchema logic for dropping nonexistent tables.
+//
+//If a table does not exist, DROP TABLE should error during preflight
+//because the statement does not change the schema as there is
+//nothing to drop.
+//In case of DROP TABLE IF EXISTS though, it should not error as this
+//is the MySQL behavior the user expects.
+func testDropNonExistentTables(t *testing.T) {
+	dropNonExistentTable := "DROP TABLE nonexistent_table;"
+	output, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("ApplySchema", "-sql", dropNonExistentTable, keyspaceName)
+	assert.NotNil(t, err)
+	assert.True(t, strings.Contains(output, "Unknown table"))
+
+	dropIfExists := "DROP TABLE IF EXISTS nonexistent_table;"
+	err = clusterInstance.VtctlclientProcess.ApplySchema(keyspaceName, dropIfExists)
+	assert.Nil(t, err)
+
+	checkTables(t, totalTableCount)
+}
+
+// checkTables check if first 2 tablets' table count match
```

Comment:
Are we guaranteed that the first tablet is always master? That is what the original test is checking on.

Reply:
Rewrite comment. Something like `checkTables checks the number of tables in the first two shards`.
Perhaps also rename this to `checkTableCounts`

Reply:
Yes, when a set of tablets are created, I always make the first one as master. I will reword the comment.